File: sdks/typescript/src/types/vectors.ts
export interface VectorEntry {
  id: string;
  embedding: number[];
  text?: string;
  metadata?: Record<string, unknown>;
}

export interface UpsertResponse {
  status: string;
  upserted: number;
}

export interface DeleteResponse {
  status: string;
  deleted: number;
}

File: sdks/python/src/bigrag/types/vectors.py
from __future__ import annotations

from typing import Any, NotRequired, TypedDict


class VectorEntry(TypedDict):
    id: str
    embedding: list[float]
    text: NotRequired[str]
    metadata: NotRequired[dict[str, Any]]


class UpsertResponse(TypedDict):
    status: str
    upserted: int


class DeleteResponse(TypedDict):
    status: str
    deleted: int
